src/main/java/com/academy/terai/Controller/ApplicationController.java

```diff
@@ -0,0 +1,59 @@
+package com.academy.terai.Controller;
+
+import com.academy.terai.Model.Application;
+import com.academy.terai.Service.ApplicationService;
+import javassist.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/application")
+public class ApplicationController {
+
+    private final ApplicationService applicationService;
+
+    @Autowired
+    public ApplicationController(ApplicationService applicationService) {
+        this.applicationService = applicationService;
+    }
+
+
+    @GetMapping
+    public ResponseEntity<List<Application>> getAllApplications() {
+        List<Application> applications = applicationService.findAll();
+        return new ResponseEntity<>(applications, HttpStatus.OK);
+    }
+
+//    @GetMapping("/{id}")
+//    public ResponseEntity<Application> getApplicationById(@PathVariable String id) throws NotFoundException {
+//        return new ResponseEntity<>(applicationService.findById(id), HttpStatus.OK);
+//    }
+
+    @GetMapping("/{id}")
+    public Optional<Application> getApplicationById(@PathVariable String id) throws NotFoundException {
```

Comment:
Not good - Optional is a Java Object used for nullable data to avoid null pointer exceptions, it's not supposed to be used as a response object.
Always return a response entity, in case it's not found - return 404 with a proper message instead.